File: z-clients/javascript/src/models/adminRoleOut.ts
// this file is @generated
import {
    type AccessRule,
    AccessRuleSerializer,
} from './accessRule';

export interface AdminRoleOut {
    id: string;
    description: string;
    rules: AccessRule[];
    policies: string[];
    context: { [key: string]: string };
    created: number;
    updated: number;
}

export const AdminRoleOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AdminRoleOut {
        return {
            id: object['id'],
            description: object['description'],
            rules: object['rules'].map((item: AccessRule) => AccessRuleSerializer._fromJsonObject(item)),
            policies: object['policies'],
            context: object['context'],
            created: object['created'],
            updated: object['updated'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AdminRoleOut): any {
        return {
            'id': self.id,
            'description': self.description,
            'rules': self.rules.map((item) => AccessRuleSerializer._toJsonObject(item)),
            'policies': self.policies,
            'context': self.context,
            'created': self.created,
            'updated': self.updated,
        };
    }
}

File: z-clients/rust/src/models/admin_role_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::access_rule::AccessRule;

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AdminRoleOut {
    pub id: String,

    pub description: String,

    pub rules: Vec<AccessRule>,

    pub policies: Vec<String>,

    pub context: std::collections::HashMap<String, String>,

    pub created: u64,

    pub updated: u64,
}

impl AdminRoleOut {
    pub fn new(
        id: String,
        description: String,
        rules: Vec<AccessRule>,
        policies: Vec<String>,
        context: std::collections::HashMap<String, String>,
        created: u64,
        updated: u64,
    ) -> Self {
        Self {
            id,
            description,
            rules,
            policies,
            context,
            created,
            updated,
        }
    }
}
